apis/facebook/api/src/facebook_api.e

```diff
@@ -148,11 +192,23 @@ feature -- Feeds: publish, delete, update
 			end
 		end
 
-	user_feed_publish_photo (a_user_id: STRING; a_photo: PATH): detachable STRING
-		local
-			l_file: RAW_FILE
+	publish_photo_on_user_feed (a_user_id: STRING; a_photo: PATH; a_params: detachable FB_USER_FEED_PUBLISHING ): detachable STRING
+		do
+			api_post_call (facebook_url (a_user_id, Void ), a_params, [a_photo, "multipart/form-data"])
+
+			if
+				attached last_response as l_response and then
```

Comment:
you should also check if there were no error.

Reply:
Yes, we have error handling in the class FACEBOOK_I, we have the interface with their documentation.
FACEBOOK_JSON is the implementation of FACEBOOK_I  with error handling and using FACEBOOK_API.

![facebookdesing](https://user-images.githubusercontent.com/23418/32508732-0861b42e-c3ca-11e7-8770-001cfa017c6a.png)